hg/hg.go

```diff
@@ -204,10 +204,11 @@ func (self *Repository) Purge() (output []byte, err error) {
 }
 
 func (self *Repository) GetLatestCommitId() (output string, err error) {
+	branch := self.makeBranchQueryFragment()
 	include := self.makeIncludeQueryFragment()
 	exclude := self.makeExcludeQueryFragment()
 	tagFilter := self.maybeTagFilter()
-	revSet := fmt.Sprintf("last((((%s) - (%s)) & %s) - desc('[ci skip]'))", include, exclude, tagFilter)
+	revSet := fmt.Sprintf("last((((%s) - (%s)) & branch(%s) & %s) - desc('[ci skip]'))", include, exclude, branch, tagFilter)
```

Comment:
Okay. Should I add parentheses and quotes to the name of the branches in `test_it_checks_correct_branch`?